ui/StatusQ/src/networkchecker.cpp

```diff
@@ -0,0 +1,62 @@
+#include "StatusQ/networkchecker.h"
+
+NetworkChecker::NetworkChecker(QObject* parent)
+    : QObject(parent)
+    , online(false)
+{
+    connect(&manager, &QNetworkAccessManager::finished, this, &NetworkChecker::onFinished);
+    connect(&timer, &QTimer::timeout, this, &NetworkChecker::checkNetwork);
+
+    updateRegularCheck(active);
+}
+
+bool NetworkChecker::isOnline() const
+{
+    return online;
+}
+
+void NetworkChecker::checkNetwork()
+{
+    QNetworkRequest request(QUrl("http://example.com"));
+    manager.get(request);
+}
+
+void NetworkChecker::onFinished(QNetworkReply* reply)
```

Comment:
It can take up to 70 sec to receive the signal, depending on the OS timeout. In this time the user will probably have no idea of why the dapps are not working. WDYT of disabling the dapps button until the connection can be established?

Reply:
Answered below